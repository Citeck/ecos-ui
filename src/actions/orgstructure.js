import { createAction } from 'redux-actions';

const prefix = 'orgstructure/';

export const setSelectedPerson = createAction(prefix + 'SET_SELECTED_PERSON');
