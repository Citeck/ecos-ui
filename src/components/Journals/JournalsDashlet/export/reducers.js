import journals from '../../../../reducers/journals';

export const reducers = {
  journals
};
