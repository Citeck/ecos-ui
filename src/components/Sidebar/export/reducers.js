import { combineReducers } from 'redux';

import app from '../../../reducers/app';
import menu from '../../../reducers/menu';
import slideMenu from '../../../reducers/slideMenu';
import view from '../../../reducers/view';

const reducers = {
  app,
  menu,
  slideMenu,
  view
};

export default () =>
  combineReducers({
    ...reducers
  });
