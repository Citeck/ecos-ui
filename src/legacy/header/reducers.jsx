import { combineReducers } from 'redux';
import {
  CREATE_CASE_WIDGET_SET_ITEMS,
  CREATE_CASE_WIDGET_SET_IS_CASCADE,
  USER_SET_PHOTO,
  USER_SET_FULLNAME,
  SITE_MENU_SET_SITE_MENU_ITEMS,
  USER_MENU_SET_ITEMS,
  SHOW_MODAL,
  HIDE_MODAL,
  SEARCH_AUTOCOMPLETE_VISIBILITY_TOGGLE,
  SEARCH_AUTOCOMPLETE_UPDATE_RESULTS,
  SEARCH_AUTOCOMPLETE_UPDATE_DOCUMENTS_RESULTS,
  SEARCH_SET_LAST_SEARCH_INDEX,
  VIEW_SET_IS_MOBILE
} from './actions';

/* caseMenuReducer */
const caseMenuInitialState = {
  isCascade: false,
  items: []
};

Object.freeze(caseMenuInitialState);

function caseMenuReducer(state = caseMenuInitialState, action) {
  switch (action.type) {
    case CREATE_CASE_WIDGET_SET_ITEMS:
      return {
        ...state,
        items: action.payload
      };

    case CREATE_CASE_WIDGET_SET_IS_CASCADE:
      return {
        ...state,
        isCascade: action.payload
      };

    default:
      return state;
  }
}

/* userMenuReducer */
const userMenuInitialState = {
  items: []
};

Object.freeze(userMenuInitialState);

function userMenuReducer(state = userMenuInitialState, action) {
  switch (action.type) {
    case USER_MENU_SET_ITEMS:
      return {
        ...state,
        items: action.payload
      };

    default:
      return state;
  }
}

/* userReducer */
const userInitialState = {
  fullName: '',
  photo: ''
};

Object.freeze(userInitialState);

function userReducer(state = userInitialState, action) {
  switch (action.type) {
    case USER_SET_FULLNAME:
      return {
        ...state,
        fullName: action.payload
      };

    case USER_SET_PHOTO:
      return {
        ...state,
        photo: action.payload
      };

    default:
      return state;
  }
}

/* siteMenuReducer */
const siteMenuInitialState = {
  items: []
};

Object.freeze(siteMenuInitialState);

function siteMenuReducer(state = siteMenuInitialState, action) {
  switch (action.type) {
    case SITE_MENU_SET_SITE_MENU_ITEMS:
      return {
        ...state,
        items: action.payload
      };

    default:
      return state;
  }
}

/* modalReducer */
const modalInitialState = {
  isOpen: false,
  title: '',
  content: '',
  onCloseCallback: null,
  buttons: []
};

/*
Button list example:
buttons: [
    {
        label: 'Button label',
        onClick: () => {
            ...some actions
        },
        bsStyle: "primary",
        isCloseButton: false
    },
    ...other buttons
]
 */

Object.freeze(modalInitialState);

function modalReducer(state = modalInitialState, action) {
  switch (action.type) {
    case SHOW_MODAL:
      return {
        ...state,
        isOpen: true,
        ...action.payload
      };

    case HIDE_MODAL:
      return modalInitialState;

    default:
      return state;
  }
}

/* searchReducer */
const searchInitialState = {
  lastSearchIndex: null,
  autocomplete: {
    isVisible: false,
    documents: {
      hasMoreRecords: false,
      items: []
    },
    sites: {
      items: []
    },
    people: {
      items: []
    }
  }
};

Object.freeze(searchInitialState);

function searchReducer(state = searchInitialState, action) {
  const payload = action.payload;

  switch (action.type) {
    case SEARCH_AUTOCOMPLETE_VISIBILITY_TOGGLE:
      return {
        ...state,
        autocomplete: {
          ...state.autocomplete,
          isVisible: action.payload
        }
      };

    case SEARCH_AUTOCOMPLETE_UPDATE_RESULTS:
      return {
        ...state,
        autocomplete: {
          ...state.autocomplete,
          documents: {
            hasMoreRecords: payload.documents.hasMoreRecords,
            items: payload.documents.items
          },
          sites: {
            items: payload.sites.items
          },
          people: {
            items: payload.people.items
          }
        }
      };

    case SEARCH_AUTOCOMPLETE_UPDATE_DOCUMENTS_RESULTS:
      return {
        ...state,
        autocomplete: {
          ...state.autocomplete,
          documents: {
            hasMoreRecords: payload.hasMoreRecords,
            items: state.autocomplete.documents.items.concat(payload.items)
          }
        }
      };

    case SEARCH_SET_LAST_SEARCH_INDEX:
      return {
        ...state,
        lastSearchIndex: payload
      };

    default:
      return state;
  }
}

/* viewReducer */
const viewInitialState = {
  isMobile: false
};

Object.freeze(viewInitialState);

function viewReducer(state = viewInitialState, action) {
  switch (action.type) {
    case VIEW_SET_IS_MOBILE:
      return {
        ...state,
        isMobile: action.payload
      };

    default:
      return state;
  }
}

/* root reducer */
export default combineReducers({
  caseMenu: caseMenuReducer,
  siteMenu: siteMenuReducer,
  modal: modalReducer,
  search: searchReducer,
  userMenu: userMenuReducer,
  user: userReducer,
  view: viewReducer
});
