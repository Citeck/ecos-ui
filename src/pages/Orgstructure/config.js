export const ORGSTRUCTURE_CONFIG = {
  id: 'orgstructure-dashboard',
  typeRef: 'emodel/type@person',
  appliedToRef: '',
  authority: null,
  priority: 0,
  config: {
    layouts: [
      {
        id: 'layout_bb6e6685-f802-4680-8606-007a42a8c1bd',
        tab: {
          label: 'Информация',
          idLayout: 'layout_bb6e6685-f802-4680-8606-007a42a8c1bd'
        },
        type: 'adaptive',
        columns: [
          {
            widgets: [
              {
                dndId: '127b05b7-8c32-4dcd-a55c-6dfbf87591b4',
                name: 'user-profile',
                label: 'dashboard-settings.widget.user-basic-info',
                id: '221ba692-9219-484d-849e-9462f7c93ec4',
                props: {
                  id: '221ba692-9219-484d-849e-9462f7c93ec4',
                  config: {}
                },
                description: 'Информация'
              },
              {
                name: 'properties',
                label: 'dashboard-settings.widget.properties',
                dndId: '4e44ef4b-b03a-486b-9a22-b6b9d3fbe374',
                id: '50d79720-f007-43f7-a91f-f7094f928b07',
                props: {
                  id: '50d79720-f007-43f7-a91f-f7094f928b07',
                  config: {
                    formId: 'uiserv/form@person_general',
                    titleAsFormName: true
                  }
                },
                description: 'Информация'
              },
              {
                dndId: '30c96b89-587b-4a44-afbc-fd5a3f8b1632',
                name: 'properties',
                label: 'dashboard-settings.widget.properties',
                id: '54b3425a-3798-4c26-a770-275d93039cf3',
                props: {
                  id: '54b3425a-3798-4c26-a770-275d93039cf3',
                  config: {
                    formId: 'uiserv/form@person_contacts',
                    titleAsFormName: true
                  }
                },
                description: 'Информация'
              },
              {
                name: 'properties',
                label: 'dashboard-settings.widget.properties',
                dndId: 'd3b6aaf0-61c5-48dd-b06f-0b3b1386f09d',
                id: 'ca2bb8ff-3e0b-414f-9e85-428df7d1458d',
                props: {
                  id: 'ca2bb8ff-3e0b-414f-9e85-428df7d1458d',
                  config: {
                    formId: 'uiserv/form@person_other',
                    titleAsFormName: true
                  }
                },
                description: 'Информация'
              },
              {
                name: 'properties',
                label: 'dashboard-settings.widget.properties',
                dndId: '72f7a248-1762-421b-975a-65be00f4d02b',
                id: 'f3796002-5214-47b8-9014-ce87cffe74c5',
                props: {
                  id: 'f3796002-5214-47b8-9014-ce87cffe74c5',
                  config: {
                    formId: 'uiserv/form@cm_person_user_groups',
                    titleAsFormName: true
                  }
                },
                description: 'Информация'
              },
              {
                name: 'documents',
                label: 'dashboard-settings.widget.documents',
                dndId: '60dc0b6d-53ee-43b3-bea6-f6c2a1265da9',
                id: '128e12a3-aa42-401c-b33b-b4850bc05821',
                props: {
                  id: '128e12a3-aa42-401c-b33b-b4850bc05821',
                  config: {
                    types: [],
                    isLoadChecklist: true
                  }
                },
                description: 'Информация'
              }
            ]
          }
        ]
      },
      {
        id: 'layout_2e5973c9-de99-48ca-8bda-b33590106022',
        tab: {
          label: 'Документы',
          idLayout: 'layout_2e5973c9-de99-48ca-8bda-b33590106022'
        },
        type: 'adaptive',
        columns: [
          {
            widgets: [
              {
                name: 'documents',
                label: 'dashboard-settings.widget.documents',
                dndId: '7f75ec6e-114b-49d3-83f8-3f8a0ea07e83',
                id: 'e8ac84e1-c36b-46a1-87c8-c04a70d98dd5',
                props: {
                  id: 'e8ac84e1-c36b-46a1-87c8-c04a70d98dd5',
                  config: {}
                },
                description: 'Документы'
              }
            ]
          }
        ]
      }
    ],
    mobile: [
      {
        id: 'layout_bb6e6685-f802-4680-8606-007a42a8c1bd',
        tab: {
          label: 'Информация',
          idLayout: 'layout_bb6e6685-f802-4680-8606-007a42a8c1bd'
        },
        type: 'mobile',
        columns: [
          {
            widgets: [
              {
                dndId: '127b05b7-8c32-4dcd-a55c-6dfbf87591b4',
                name: 'user-profile',
                label: 'dashboard-settings.widget.user-basic-info',
                id: '221ba692-9219-484d-849e-9462f7c93ec4',
                props: {
                  id: '221ba692-9219-484d-849e-9462f7c93ec4',
                  config: {}
                },
                description: 'Информация'
              },
              {
                name: 'properties',
                label: 'dashboard-settings.widget.properties',
                dndId: '4e44ef4b-b03a-486b-9a22-b6b9d3fbe374',
                id: '50d79720-f007-43f7-a91f-f7094f928b07',
                props: {
                  id: '50d79720-f007-43f7-a91f-f7094f928b07',
                  config: {
                    formId: 'uiserv/form@person_general',
                    titleAsFormName: true
                  }
                },
                description: 'Информация'
              },
              {
                dndId: '30c96b89-587b-4a44-afbc-fd5a3f8b1632',
                name: 'properties',
                label: 'dashboard-settings.widget.properties',
                id: '54b3425a-3798-4c26-a770-275d93039cf3',
                props: {
                  id: '54b3425a-3798-4c26-a770-275d93039cf3',
                  config: {
                    formId: 'uiserv/form@person_contacts',
                    titleAsFormName: true
                  }
                },
                description: 'Информация'
              },
              {
                name: 'properties',
                label: 'dashboard-settings.widget.properties',
                dndId: 'd3b6aaf0-61c5-48dd-b06f-0b3b1386f09d',
                id: 'ca2bb8ff-3e0b-414f-9e85-428df7d1458d',
                props: {
                  id: 'ca2bb8ff-3e0b-414f-9e85-428df7d1458d',
                  config: {
                    formId: 'uiserv/form@person_other',
                    titleAsFormName: true
                  }
                },
                description: 'Информация'
              },
              {
                name: 'properties',
                label: 'dashboard-settings.widget.properties',
                dndId: '72f7a248-1762-421b-975a-65be00f4d02b',
                id: 'f3796002-5214-47b8-9014-ce87cffe74c5',
                props: {
                  id: 'f3796002-5214-47b8-9014-ce87cffe74c5',
                  config: {
                    formId: 'uiserv/form@cm_person_user_groups',
                    titleAsFormName: true
                  }
                },
                description: 'Информация'
              },
              {
                name: 'documents',
                label: 'dashboard-settings.widget.documents',
                dndId: '60dc0b6d-53ee-43b3-bea6-f6c2a1265da9',
                id: '128e12a3-aa42-401c-b33b-b4850bc05821',
                props: {
                  id: '128e12a3-aa42-401c-b33b-b4850bc05821',
                  config: {}
                },
                description: 'Информация'
              }
            ]
          }
        ]
      },
      {
        id: 'layout_2e5973c9-de99-48ca-8bda-b33590106022',
        tab: {
          label: 'Документы',
          idLayout: 'layout_2e5973c9-de99-48ca-8bda-b33590106022'
        },
        type: 'mobile',
        columns: [
          {
            widgets: [
              {
                name: 'documents',
                label: 'dashboard-settings.widget.documents',
                dndId: '7f75ec6e-114b-49d3-83f8-3f8a0ea07e83',
                id: 'e8ac84e1-c36b-46a1-87c8-c04a70d98dd5',
                props: {
                  id: 'e8ac84e1-c36b-46a1-87c8-c04a70d98dd5',
                  config: {}
                },
                description: 'Документы'
              }
            ]
          }
        ]
      }
    ],
    version: 'v2',
    v2: {
      widgets: [
        {
          dndId: '127b05b7-8c32-4dcd-a55c-6dfbf87591b4',
          name: 'user-profile',
          label: 'dashboard-settings.widget.user-basic-info',
          id: '221ba692-9219-484d-849e-9462f7c93ec4',
          props: {
            id: '221ba692-9219-484d-849e-9462f7c93ec4',
            config: {
              widgetDisplayCondition: null,
              elementsDisplayCondition: {}
            }
          },
          description: 'Информация'
        },
        {
          name: 'properties',
          label: 'dashboard-settings.widget.properties',
          dndId: '4e44ef4b-b03a-486b-9a22-b6b9d3fbe374',
          id: '50d79720-f007-43f7-a91f-f7094f928b07',
          props: {
            id: '50d79720-f007-43f7-a91f-f7094f928b07',
            config: {
              widgetDisplayCondition: null,
              elementsDisplayCondition: {},
              formId: 'uiserv/form@person_general',
              titleAsFormName: true
            }
          },
          description: 'Информация'
        },
        {
          dndId: '30c96b89-587b-4a44-afbc-fd5a3f8b1632',
          name: 'properties',
          label: 'dashboard-settings.widget.properties',
          id: '54b3425a-3798-4c26-a770-275d93039cf3',
          props: {
            id: '54b3425a-3798-4c26-a770-275d93039cf3',
            config: {
              widgetDisplayCondition: null,
              elementsDisplayCondition: {},
              formId: 'uiserv/form@person_contacts',
              titleAsFormName: true
            }
          },
          description: 'Информация'
        },
        {
          name: 'properties',
          label: 'dashboard-settings.widget.properties',
          dndId: 'd3b6aaf0-61c5-48dd-b06f-0b3b1386f09d',
          id: 'ca2bb8ff-3e0b-414f-9e85-428df7d1458d',
          props: {
            id: 'ca2bb8ff-3e0b-414f-9e85-428df7d1458d',
            config: {
              widgetDisplayCondition: null,
              elementsDisplayCondition: {},
              formId: 'uiserv/form@person_other',
              titleAsFormName: true
            }
          },
          description: 'Информация'
        },
        {
          name: 'properties',
          label: 'dashboard-settings.widget.properties',
          dndId: '72f7a248-1762-421b-975a-65be00f4d02b',
          id: 'f3796002-5214-47b8-9014-ce87cffe74c5',
          props: {
            id: 'f3796002-5214-47b8-9014-ce87cffe74c5',
            config: {
              widgetDisplayCondition: null,
              elementsDisplayCondition: {},
              formId: 'uiserv/form@cm_person_user_groups',
              titleAsFormName: true
            }
          },
          description: 'Информация'
        },
        {
          name: 'documents',
          label: 'dashboard-settings.widget.documents',
          dndId: '60dc0b6d-53ee-43b3-bea6-f6c2a1265da9',
          id: '128e12a3-aa42-401c-b33b-b4850bc05821',
          props: {
            id: '128e12a3-aa42-401c-b33b-b4850bc05821',
            config: {
              widgetDisplayCondition: null,
              elementsDisplayCondition: {},
              types: [],
              isLoadChecklist: true
            }
          },
          description: 'Информация'
        },
        {
          name: 'documents',
          label: 'dashboard-settings.widget.documents',
          dndId: '7f75ec6e-114b-49d3-83f8-3f8a0ea07e83',
          id: 'e8ac84e1-c36b-46a1-87c8-c04a70d98dd5',
          props: {
            id: 'e8ac84e1-c36b-46a1-87c8-c04a70d98dd5',
            config: {
              widgetDisplayCondition: null,
              elementsDisplayCondition: {}
            }
          },
          description: 'Документы'
        }
      ],
      desktop: [
        {
          id: 'layout_bb6e6685-f802-4680-8606-007a42a8c1bd',
          tab: {
            label: 'Информация',
            idLayout: 'layout_bb6e6685-f802-4680-8606-007a42a8c1bd'
          },
          type: 'adaptive',
          columns: [
            {
              widgets: [
                '221ba692-9219-484d-849e-9462f7c93ec4',
                '50d79720-f007-43f7-a91f-f7094f928b07',
                '54b3425a-3798-4c26-a770-275d93039cf3',
                'ca2bb8ff-3e0b-414f-9e85-428df7d1458d',
                'f3796002-5214-47b8-9014-ce87cffe74c5',
                '128e12a3-aa42-401c-b33b-b4850bc05821'
              ]
            }
          ]
        },
        {
          id: 'layout_2e5973c9-de99-48ca-8bda-b33590106022',
          tab: {
            label: 'Документы',
            idLayout: 'layout_2e5973c9-de99-48ca-8bda-b33590106022'
          },
          type: 'adaptive',
          columns: [
            {
              widgets: ['e8ac84e1-c36b-46a1-87c8-c04a70d98dd5']
            }
          ]
        }
      ],
      mobile: [
        {
          id: 'layout_bb6e6685-f802-4680-8606-007a42a8c1bd',
          tab: {
            label: 'Информация',
            idLayout: 'layout_bb6e6685-f802-4680-8606-007a42a8c1bd'
          },
          type: 'mobile',
          columns: [
            {
              widgets: [
                '221ba692-9219-484d-849e-9462f7c93ec4',
                '50d79720-f007-43f7-a91f-f7094f928b07',
                '54b3425a-3798-4c26-a770-275d93039cf3',
                'ca2bb8ff-3e0b-414f-9e85-428df7d1458d',
                'f3796002-5214-47b8-9014-ce87cffe74c5',
                '128e12a3-aa42-401c-b33b-b4850bc05821'
              ]
            }
          ]
        },
        {
          id: 'layout_2e5973c9-de99-48ca-8bda-b33590106022',
          tab: {
            label: 'Документы',
            idLayout: 'layout_2e5973c9-de99-48ca-8bda-b33590106022'
          },
          type: 'mobile',
          columns: [
            {
              widgets: ['e8ac84e1-c36b-46a1-87c8-c04a70d98dd5']
            }
          ]
        }
      ]
    }
  },
  attributes: {}
};

export const ATTRIBUTES = [
  { id: 'authorityGroups', name: 'Группы', type: 'ASSOC' },

  { id: 'firstName', name: 'Имя', type: 'TEXT' },

  { id: 'lastName', name: 'Фамилия', type: 'TEXT' },

  { id: 'middleName', name: 'Отчество', type: 'TEXT' },

  { id: 'email', name: 'EMail', type: 'TEXT' },

  { id: 'jobTitle', name: 'Должность', type: 'TEXT' },

  { id: 'birthDate', name: 'Дата рождения', type: 'DATE' },
  { id: 'birthMonthDay', name: 'Birth MonthDay', type: 'NUMBER' },
  { id: 'sex', name: 'Пол', type: 'TEXT' },
  { id: 'photo', name: 'Фото', type: 'CONTENT' },

  { id: 'city', name: 'Город', type: 'TEXT' },

  { id: 'organization', name: 'Компания', type: 'TEXT' },

  { id: 'nameInGenitiveCase', name: 'ФИО в родительном падеже', type: 'TEXT' },

  { id: 'typeOfEmployment', name: 'Employment type', type: 'TEXT' },

  { id: 'phoneInternal', name: 'Внутренний телефон', type: 'TEXT' },

  { id: 'phoneWorking', name: 'Рабочий телефон', type: 'TEXT' },

  { id: 'personDisabled', name: 'Учетная запись отключена', type: 'BOOLEAN' },

  { id: 'personDisableReason', name: 'Причина отключения учетной записи', type: 'TEXT' },

  { id: 'countryCode', name: 'Код страны', type: 'TEXT' },

  { id: 'timezone', name: 'Часовой пояс', type: 'TEXT' },

  { id: 'itn', name: 'ИНН', type: 'TEXT' },

  { id: 'inila', name: 'СНИЛС', type: 'TEXT' },

  { id: 'skype', name: 'Скайп', type: 'TEXT' },

  { id: 'mobile', name: 'Телефон мобильный', type: 'TEXT' },

  { id: 'location', name: 'Местоположение', type: 'TEXT' },

  { id: 'userStatus', name: 'Статус', type: 'TEXT' },

  { id: 'workingCalendar', name: 'Рабочий календарь', type: 'ASSOC' },

  { id: 'atWorkplace', name: 'На рабочем месте', type: 'BOOLEAN' }
];
