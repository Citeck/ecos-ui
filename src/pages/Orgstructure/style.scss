.app-content.orgstructure-page {
  overflow: hidden;
}
.orgstructure-page {
  &__grid {
    &-container {
      display: grid;
      grid-template-columns: minmax(300px, 1fr) 3fr;
      grid-template-rows: 1fr;
      grid-column-gap: 24px;
      grid-row-gap: 0px;
      height: 100%;
    }

    &-main {
      position: relative;
      margin-top: 23px;
      margin-left: 12px;
      width: 100%;
      height: 100%;
      overflow: hidden;
      background-color: white;
      border-radius: 12px;
    }

    &-layout {
      display: block;
      position: relative;
      padding-top: 23px;
      overflow-y: scroll;

      .ecos-layout {
        padding: 0 18px 0 4px;
      }
    }

    &-widgets {
      display: grid;
      grid-template-columns: repeat(auto-fit, minmax(290px, 1fr));
      grid-gap: 10px;
      padding-right: 10px;
      overflow: auto;

      &-block {
        display: flex;
        gap: 10px;
        flex-direction: column;
        padding-bottom: 20px;
      }
    }

    &-empty-widgets {
      display: flex;
      justify-content: center;
      align-items: center;
      font-weight: 700;
      color: #767676;
    }
  }
}
