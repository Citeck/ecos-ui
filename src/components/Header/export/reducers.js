import { combineReducers } from 'redux';

import iconSelect from '../../../reducers/iconSelect';
import header from '../../../reducers/header';
import menu from '../../../reducers/menu';
import menuSettings from '../../../reducers/menuSettings';
import user from '../../../reducers/user';
import view from '../../../reducers/view';

const reducers = {
  iconSelect,
  header,
  menu,
  menuSettings,
  user,
  view
};

export default () =>
  combineReducers({
    ...reducers
  });
