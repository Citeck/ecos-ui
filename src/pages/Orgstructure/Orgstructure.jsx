import React from 'react';
import { connect } from 'react-redux';

import Structure from './components/Structure';
import Widgets from './components/Widgets';
import { setSelectedPerson } from '../../actions/orgstructure';
import { getSearchParams } from '../../helpers/urls';

import './style.scss';

class Orgstructure extends React.Component {
  componentDidMount() {
    const { onSelectPerson } = this.props;
    const { recordRef } = getSearchParams() || {};

    if (recordRef) {
      onSelectPerson(recordRef);
    }
  }

  render() {
    return (
      <div className="orgstructure-page__grid__container">
        <div className="orgstructure-page__grid__main">
          <Structure />
        </div>
        <Widgets />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  onSelectPerson: recordRef => dispatch(setSelectedPerson(recordRef))
});

export default connect(
  null,
  mapDispatchToProps
)(Orgstructure);
