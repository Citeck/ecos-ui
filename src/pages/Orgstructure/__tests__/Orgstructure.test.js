import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { unmountComponentAtNode } from 'react-dom';
import { Provider } from 'react-redux';

import Orgstructure from '../Orgstructure';
import { initialState } from '../../../reducers/orgstructure';
import configureStore from '../../../store';

configure({ adapter: new Adapter() });

const renderWithRedux = (component, initialState) => {
  const store = configureStore(null, { orgstructure: initialState });

  return (
    <Provider store={store}>
      <Orgstructure />
    </Provider>
  );
};
let container = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('<Orgstructure />', () => {
  it('render without selected recordRef', () => {
    const component = shallow(renderWithRedux(<Orgstructure />, initialState));

    expect(component).toMatchSnapshot();
  });

  it('render with selected recordRef', () => {
    const component = shallow(renderWithRedux(<Orgstructure />, { id: 'emodel/person@bryzadmin' }));

    expect(component).toMatchSnapshot();
  });
});
