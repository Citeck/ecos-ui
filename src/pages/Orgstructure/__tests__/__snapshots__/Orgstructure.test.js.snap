// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Orgstructure /> render with selected recordRef 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "asyncReducers": Object {},
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(Orgstructure) />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "store": Object {
      "asyncReducers": Object {},
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    },
    "storeSubscription": null,
  },
}
`;

exports[`<Orgstructure /> render without selected recordRef 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "asyncReducers": Object {},
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(Orgstructure) />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "store": Object {
      "asyncReducers": Object {},
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    },
    "storeSubscription": null,
  },
}
`;
